### ISSUE TITLE
Python Sklearn Type XGBoostClassifier doesn't save num_class

### ISSUE THREAD

**Issue by sushanttripathy** [dmlc/xgboost#817]
After I train an XGBoost model using GridSearchCV and pickle.dump it into a file. The pickle.load(ed) classifier issues the following error on using either predict or predict_proba.

Parameter num_class is not presented

I can post the code examples if needed!

**Comment by namp** [comment]
I'm also confronted with the exact same problem!

**Comment by namp** [comment]
I think I found a "solution" by explicitly setting:

cl_loaded._Booster.set_param({'num_class': N})

where N is the number of classes.

Still, I believe this is a bug though...

**Comment by mswastik** [comment]
I am also not able to use softmax objective, looked around and found there is no mention of "multi:softprob" in XGBoost's sklearn wrapper https://github.com/dmlc/xgboost/blob/master/python-package/xgboost/sklearn.py.

@namp when using XGBoostClassifier there is no way to set 'num_class' parameter for "multi:softprob" objective. I am using ipython, can that be the reason.
